### ISSUE TITLE
Repeatedly tries to send ServerRequestRegisterOpen after disableTracking(true)

### ISSUE THREAD

**Issue by rhoglund** [BranchMetrics/android-branch-deep-linking-attribution#973]
After bumping to 5.1.x we noticed:

If we disable tracking, and then reopen the app, the app just keeps retrying to send the ServerRequestRegisterOpen.

Log very quickly becomes filled with:

I/BranchSDK: processNextQueueItem, req ServerRequestRegisterOpen
I/BranchSDK: processNextQueueItem, req ServerRequestRegisterOpen
I/BranchSDK: processNextQueueItem, req ServerRequestRegisterOpen
...

If we call enable tracking and disable it again it stops, until next reopen. Looking at the code when disableTracking is called, it calls 'onTrackingDisabled(context)' which will clear pending requests - which makes it stop. However when initializing, this is not called - only updateTrackingState, which does not clear requests.

**Comment by gdeluna-branch** [comment]
I will take a look at this.

**Comment by gdeluna-branch** [comment]
Have the fix, having it reviewed and to be released soon. Thanks.

**Comment by chris-prop** [comment]
I'm not sure if your fix addresses this, but after a crash, I was getting repeated calls to

```
I/BranchSDK: processNextQueueItem, req ServerRequestRegisterInstall
```
The issue persisted after closing/restarting, but uninstall/reinstall fixed it.

**Comment by gdeluna-branch** [comment]
Yep there's a fix is in 5.1.**2** to be released today, once it is published through maven. It was introduced in 5.1.0 because of an oversight in retrying requests when they should have been considered un-retryable. This request was not set to retry initially.

**Comment by gdeluna-branch** [comment]
Fix released in 5.1.2 
https://github.com/BranchMetrics/android-branch-deep-linking-attribution/releases/tag/5.1.2
https://search.maven.org/artifact/io.branch.sdk.android/library/5.1.2/aar

Please reopen if this still repros.

Thanks
